src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,172 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+    private Node<T> first;
+    private Node<T> last;
+
+    private static class Node<T> {
+        private T item;
+        private Node<T> next;
+        private Node<T> prev;
+
+        public Node(Node<T> prev, T item, Node<T> next) {
+            this.prev = prev;
+            this.item = item;
+            this.next = next;
+        }
+    }
+
     @Override
     public void add(T value) {
+        if (size > 0) {
+            addNodeToEnd(value);
+        } else {
+            addFirst(value);
+        }
     }
 
     @Override
     public void add(T value, int index) {
+        checkIndexForAdd(index);
+        addWithIndex(value,index);
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T obj : list) {
```

Comment:
```suggestion
        for (T element : list) {
```

not informative variable name